tests/ApiFilterTest.php

```diff
@@ -175,19 +177,50 @@ public function shouldParseQueryParametersAndApplyThemOneByOneToQueryBuilder(
         $this->assertSame($expectedPreparedValues, $preparedValues);
     }
 
+    /**
+     * @test
+     * @dataProvider provideInvalidQueryParameters
+     */
+    public function shouldNotParseInvalidQueryParameters(array $invalidQueryParameters, string $expectedMessage): void
+    {
+        $this->expectException(ApiFilterException::class);
+        $this->expectExceptionMessage($expectedMessage);
+
+        $this->apiFilter->parseFilters($invalidQueryParameters);
+    }
+
+    public function provideInvalidQueryParameters(): array
+    {
+        return [
+            // invalidQueryParameters, expectedMessage
+            'empty filter' => [
+                ['column' => ['' => 'value']],
+                'Filter "" is not implemented. For column "column" with value "value".',
+            ],
+            'unknown filter' => [
+                ['column' => ['unknown' => 'value']],
+                'Filter "unknown" is not implemented. For column "column" with value "value".',
+            ],
+            'tuples in IN filter' => [
+                ['(id, name)' => ['in' => ['(1,one)', '(2,two)']]],
+                'Tuples are not allowed in IN filter.',
```

Comment:
This test case shows that only one generic exception class is not enough. They suggest more types of errors.